content/fundamentals/unit-testing.md

```diff
@@ -160,6 +160,31 @@ Instead of using the production version of any provider, you can override it wit
 
 <app-banner-courses></app-banner-courses>
 
+#### Auto mocking
+
+Nest also allows you to define a mock factory to apply to all of your missing dependencies. This is useful for cases where you have a large number of dependencies in a class and mocking all of them will take a long time and a lot of setup. To make use of this feature, the `createTestingModule()` will need to be chained up with the `useMocker()` method, passing a factory for your dependency mocks. This factory can take in an optional token, which is an instance token, any token which is valid for a Nest provider, and returns a mock implementation. Otherwise, a general mock factory, like `createMock` from [`@golevelup/ts-jest`](https://github.com/golevelup/nestjs/tree/master/packages/testing) can be passed directly.
+
+```typescript
+describe('CatsController', () => {
+  let controller: CatsController;
+
+  beforeEach(async () => {
+    const modRef = await Test.createTestingModule({
+      controllers: [CatsController],
+    })
+    .useMocker((token) => {
+      if (token === CatsService) {
+        return { findAll: jest.fn().mockResolveValue(results) }
+      }
```

Comment:
Maybe we could show how to use `jest-mock` here?

```typescript
const moduleMocker = new ModuleMocker(global) // we can move this outside `describe()`

if (typeof token === 'function') {
  const mockMetadata = moduleMocker.getMetadata(token) as MockFunctionMetadata<any, any>
  const Mock = moduleMocker.generateFromMetadata(mockMetadata)
  return new Mock()
}
// not sure what to put here
// maybe we can also preserve the "if (token === CatsService)" clause at the beginning just to show how to the mocker more "specific"
```

Reply:
Done